specs/core/0_beacon-chain.md

```diff
@@ -1091,7 +1088,7 @@ def get_effective_balance(state: BeaconState, index: ValidatorIndex) -> Gwei:
     """
     Return the effective balance (also known as "balance at stake") for a validator with the given ``index``.
     """
-    return min(get_balance(state, index), MAX_DEPOSIT_AMOUNT)
+    return min(state.validator_registry[index].rounded_balance, MAX_DEPOSIT_AMOUNT)
```

Comment:
This function is used for reward/penalty calculations. Do we want to use rounded balance for reward/penalties?

Reply:
I'd argue so. The balance at stake is rounded so rewards/penalties should also be.

Reply:
This does effect a validator's effective rate of return as rounded_balance <= actual_balance. In the event that a validator has >= 32ETH and has always had so, then this doesn't matter. In the event that a validator's balance drops to 16ETH, however, by this makes the difference between having 64 ETH and 65 ETH after validating for a long time.

If we allow balances to grow exponentially as per #730, then this always matters. See graphs attached below:

![econ_test_init_bal16](https://user-images.githubusercontent.com/12530043/55544784-1b216000-56cc-11e9-814b-f4f67c451e3e.png)
![econ_test_init_bal32](https://user-images.githubusercontent.com/12530043/55544785-1b216000-56cc-11e9-894e-993a0bc8c3dc.png)

Reply:
Right. I guess a downside of using rounded balance for rewards is that it creates a misalignment between capital costs (not rounded) and rewards (rounded).